### DISCUSSION TITLE
Mermaid Diagrams for App Behavior Discussions

### DISCUSSION THREAD

**Discussion by codetricity** [caitlynbakery/theta_x_multi_bloc#3] [Ideas]
We may be able to use [Mermaid diagrams](https://mermaid-js.github.io/mermaid/#/stateDiagram) in the discussions about app behavior.

```mermaid
stateDiagram-v2
    First_Screen --> Video_Settings_Screen: switch to video settings bloc on second screen
```

**Comment by jcasman** [comment]
Mermaid Diagrams look useful. I will try using it later today.

Including this just for my notes:
"Mermaid is a JavaScript based diagramming and charting tool that uses Markdown-inspired text definitions and a renderer to create and modify complex diagrams. The main purpose of Mermaid is to help documentation catch up with development."

**Reply by codetricity** [reply]
> Mermaid Diagrams look useful. I will try using it later today.
> 
> Including this just for my notes: "Mermaid is a JavaScript based diagramming and charting tool that uses Markdown-inspired text definitions and a renderer to create and modify complex diagrams. The main purpose of Mermaid is to help documentation catch up with development."

This part from the site was also interesting:

_not having diagrams or docs ruins productivity and hurts organizational learning._

I'm trying to figure out a way to work collaboratively and discuss:

* what appears on each screen
*  what happens when a button is pressed

I believe that a side-effect may be to separate out the functionality of an app from the GUI design.

Although aesthetic design is super important, I feel that the tools to discuss aesthetic design are different than the tools to discuss functional design.

More information on [GitHub Flavored Markdown](https://github.github.com/gfm/)
